### PR TITLE
[New] Query feature table

### PR DESCRIPTION
**Pull request by pgruenler**
## Description

This PR implements `Query feature table` in `Features` category.
URL to README: 

## Linked Issue(s)

- `swift/issues/3577`

### CODE REVIEW

File: Shared/Samples/Feature layer query/README.md
Comment (yo1995) [inline Shared/Samples/Feature layer query/README.md:1]: Hi Phil, can you help rename the sample to "Query feature table" according to the common design? There was an effort to standardize the names across platforms, and the old iOS sample name deviates from the common design. Thanks.

- the folder name
- README title
- title in metadata
- swift filename
- view type name (FeatureLayerQueryView -> QueryFeatureTableView)

Comment (pgruenler) [reply Shared/Samples/Feature layer query/README.md:1]: Ah, I saw there were different names in the design and the old samples; I picked the wrong one. I'll make the changes.

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (des12437) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:107]:
```suggestion
        let featureTable = ServiceFeatureTable(
             url: URL(string: "https://services.arcgis.com/jIL9msH9OI208GCb/arcgis/rest/services/USA_Daytime_Population_2016/FeatureServer/0")!
)
```

**Review by des12437 — COMMENTED** [review]
A few whitespace suggestions.

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (des12437) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:108]:
```suggestion
        )
```
The indentation was off in my first suggestion.

**Review by des12437 — DISMISSED** [review]

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:73]:
```suggestion
                        let queryResultFeatures = Array(queryResult.features())
                        if !queryResultFeatures.isEmpty {
```

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:66]:
```suggestion
                    // Unselects all features.
                    model.featureLayer.clearSelection()
```

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:74]:
```suggestion
                            // Displays the selection.
```

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:67]:
```suggestion
                    // Makes the query parameters and executes the query.
```

**Comment by des12437** [comment]
Please resolve the merge conflicts.

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:41]: I was a bit surprised that the [trailing closure linter rule](https://realm.github.io/SwiftLint/trailing_closure.html) isn't triggered on this line. So it's up to you if make this change.

```suggestion
                .onSubmit(of: .search) {
                    model.currentQuery = searchBarText
                }
```

**Review by yo1995 — COMMENTED** [review]
Will approve once the merge conflict is resolved.

**Review by des12437 — APPROVED** [review]

**Review by yo1995 — APPROVED** [review]

File: Samples.xcodeproj/project.pbxproj
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:187]: These changes shouldn't be here - will be fixed in another PR.

**Pull request merged**
